nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -115,7 +122,11 @@ impl<'a> Planner<'a> {
         self.do_plan_expunge()?;
         self.do_plan_add()?;
         self.do_plan_decommission()?;
-        self.do_plan_zone_updates()?;
+        if let UpdateStepResult::ContinueToNextStep =
+            self.do_plan_mgs_updates()?
+        {
+            self.do_plan_zone_updates()?;
+        }
```

Comment:
This might be nitpicky, but it seems a little weird that if we don't get `ContinueToNextStep`, we skip _the immediate_ next step but still do the reset of planning (which admittedly is not very much). Maybe this should be specific to "can we do further updates"? (#8284 / #8285 / #8298 are all closely related, so also fine to defer this until we do some combination of them.)

Reply:
Yeah, I've been assuming we'd iterate on the exact control flow / pattern here as we add more steps.